lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestFailedTaskEventHandler.cs

```diff
@@ -84,26 +92,65 @@ public void OnNext(IDriverStarted value)
 
             public void OnNext(IAllocatedEvaluator value)
             {
-                value.SubmitTask(TaskConfiguration.ConfigurationModule
-                    .Set(TaskConfiguration.Identifier, TaskId)
-                    .Set(TaskConfiguration.Task, GenericType<FailTask>.Class)
-                    .Build());
+                value.SubmitTask(GetTaskConfiguration());
             }
 
             public void OnNext(IFailedTask value)
             {
-                Logger.Log(Level.Error, FailedTaskMessage);
+                _numFailedTasksReceived++;
+
+                if (value.Id != TaskId)
+                {
+                    throw new Exception("Received Task ID " + value.Id + " instead of the expected Task ID " + TaskId);
+                }
+
                 if (value.Message == null || value.Message != ExpectedExceptionMessage)
                 {
                     throw new Exception("Exception message not properly propagated. Received message " + value.Message);
                 }
 
-                if (value.Id != TaskId)
+                if (_shouldReceiveSerializableException)
                 {
-                    throw new Exception("Received Task ID " + value.Id + " instead of the expected Task ID " + TaskId);
+                    if (_numFailedTasksReceived == NumFailedTasksExpected)
+                    {
+                        Logger.Log(Level.Error, FailedTaskMessage);
+                    }
+
+                    if (value.AsError() == null || !(value.AsError() is TestSerializableException))
+                    {
+                        throw new Exception("Exception should have been serialized properly.");
+                    }
+
+                    if (value.AsError().Message != ExpectedExceptionMessage)
+                    {
+                        throw new Exception("Incorrect Exception message, got message: " + value.AsError().Message);
+                    }
+
+                    value.GetActiveContext().Value.Dispose();
+                }
+                else
+                {
+                    var taskException = value.AsError();
```

Comment:
Cause is not a method in the `Interface`. I think instead I should just remove it.